halyard-cli/src/test/groovy/com/netflix/spinnaker/halyard/cli/command/v1/CommandTreeSpec.groovy

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1
+
+import com.netflix.spinnaker.halyard.cli.command.v1.AdminCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.BackupCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.ConfigCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.DeployCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.HalCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.NestableCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.TaskCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.VersionCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.DeploymentEnvironmentCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.EditConfigCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.FeaturesCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.GenerateCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.MetricStoresCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.PersistentStorageCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.SecurityCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.VersionConfigCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.ci.CiCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.providers.ProviderCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.api.ApiSecurityCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.AuthnCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.oauth2.EditOAuth2Command
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.oauth2.OAuth2Command
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.saml.EditSamlCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.saml.SamlCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authz.AuthzCommand
+import com.netflix.spinnaker.halyard.cli.command.v1.config.security.ui.UiSecurityCommand
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.AuthnMethodEnableDisableCommandBuilder.AuthnMethodEnableDisableCommand
+
+class CommandTreeSpec extends Specification {
+
+  /**
+   * Documents and validates the tree structure of the hal command and all of its subcommands.
```

Comment:
Awesome. Thank you. Ready any time.

Two things to call out:
* Next time I'm in New York I'd like to buy you a beer or alternate beverage of your choice. #appreciation 
* Please be vigilant and frank for when the tests feel heavy to you. Let's nip that problem in the bud when it comes up.